salt/minion.py

```diff
@@ -2943,19 +2978,20 @@ def _post_master_init(self, master):
         '''
         log.debug("subclassed _post_master_init")
 
-        self.opts['master'] = master
+        if self.connected:
```

Comment:
I'm not sure, but this line is in the Minion._post_master_init as well, so I was guessing that there is some edge condition & left it in.